src/core/create_compilers/RollupCompiler.ts

```diff
@@ -154,11 +156,51 @@ export default class RollupCompiler {
 
 		const that = this;
 
-		// TODO this is hacky. refactor out into an external rollup plugin
-		(mod.plugins || (mod.plugins = [])).push(css_chunks({ injectImports: true }));
-		if (!/[\\/]client\./.test(entry_point)) {
-			return mod;
-		}
+		/**
+		 * Track css code hashes for proper chunk hashing
+		 */
+		const css_hashing = () => {
+			const css_hashes: Record<string, string> = {};
+			return <Plugin>{
+				name: 'sapper-css-hashing',
```

Comment:
Can this plugin be combined with the `sapper-css-injection` plugin? It seems like the hashing is really necessary because of what that plugin is doing and isn't something we'd use in a standalone fashion

Reply:
It needs to be before the css chunk plugin and is also critical for the client chunk.

Reply:
🤔 Actually this exact issue should also apply to `rollup-plugin-css-chunks` as there is also some `chunk.code` stuff happening without a corresponding `augmentChunkHash` hook. So `sapper-css-hashing` should actually fix this for the css chunk plugin as well as the sapper rollup compiler.

Reply:
I was hoping to combine the `sapper-css-injection` plugin with `rollup-plugin-css-chunk`: https://github.com/domingues/rollup-plugin-css-chunks/pull/8. There's no reason that our css injection functionality couldn't be used by other Rollup projects, so I think it'd better live as a project separate from Sapper

Reply:
Thinking about it this maybe could have been solved on `rollup-plugin-css-chunks` level. I already tried to update to the newest version, but that had a major update in the mean time. Using the newest version in sapper had noticeable side effects. Fix it there might still need extra work on sapper side, I guess.

Reply:
I just sent https://github.com/sveltejs/sapper/pull/1666 to update to the latest `rollup-plugin-css-chunks`. I had to bump Rollup as well to make it work